Sources/TuistCache/Cache/CacheStoring.swift

```diff
@@ -1,23 +1,29 @@
 import Foundation
 import RxSwift
 import TSCBasic
+import TuistCore
 
-public protocol CacheStoraging {
+public protocol CacheStoring {
     /// Returns if the target with the given hash exists in the cache.
-    /// - Parameter hash: Target's hash.
+    /// - Parameters:
+    ///   - hash: Target's hash.
+    ///   - userConfig: The user configuration.
     /// - Returns: An observable that returns a boolean indicating whether the target is cached.
-    func exists(hash: String) -> Single<Bool>
+    func exists(hash: String, config: Config) throws -> Single<Bool>
```

Comment:
Why do the methods throw? Since we are returning an observable primitive, the error should be thrown when subscribing to it.

```swift
exists(hash: "xxx").subscribe(onError: { error in })
```

Reply:
You're right... good point. I changed that